pkg/sql/opt/indexrec/index_recommendation_set.go

```diff
@@ -295,39 +322,102 @@ func (ir *indexRecommendation) storingColumns() []tree.Name {
 	return storingCols
 }
 
-// indexRecommendationString returns the string output for an index
-// recommendation, containing the SQL command(s) needed to follow this
-// recommendation.
-func (ir *indexRecommendation) indexRecommendationString(
-	indexCols []tree.IndexElem, storing []tree.Name,
-) string {
-	var sb strings.Builder
-	tableName := tree.NewUnqualifiedTableName(ir.index.tab.Name())
-
-	var dropCmd tree.DropIndex
-	unique := false
-	if ir.existingIndex != nil {
-		sb.WriteString("   SQL commands: ")
-		indexName := tree.UnrestrictedName(ir.existingIndex.Name())
-		dropCmd.IndexList = []*tree.TableIndexName{{Table: *tableName, Index: indexName}}
-
-		// Maintain uniqueness if the existing index is unique.
-		unique = ir.existingIndex.IsUnique()
-	} else {
-		sb.WriteString("   SQL command: ")
-	}
+// outputCommand is an interface that defines methods that an index
+// recommendation type has to implement. This is a helper to print the final
+// index recommendations.
+type outputCommand interface {
```

Comment:
I think this interface is more complicated than the solution I was working toward in the collab session. I'll try to finish up that PR tomorrow morning.

Reply:
I have a PR up that should simplify things for you: #86754. It adds a `Replacement bool` to the output recommendations to signify the type of recommendation. If you're adding another type, you can use an enum (like [this](https://github.com/cockroachdb/cockroach/blob/6e191d5f91508102ee44e97485ea4358b0c10375/pkg/sql/opt/optbuilder/scope.go#L115-L138)).

Reply:
Thanks for the fix! Looking at the pr now.